pkg/kanctl/profile.go

```diff
@@ -234,26 +233,59 @@ func getLocationParams(cmd *cobra.Command) (*locationParams, error) {
 	}, nil
 }
 
-func constructProfile(lP *locationParams, secret *v1.Secret, role string) *v1alpha1.Profile {
+func constructProfile(lP *locationParams, secret *v1.Secret) *v1alpha1.Profile {
```

Comment:
I noticed that there are no unit tests for testing either constructProfile/constructSecret or even the caller createNewProfile(used in newAzureProfileCmd/newGCPProfileCmd/newS3CompliantProfileCmd). It would be useful to add tests using fake kube clients.